examples/deployContractUDC/main.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/starknet.go/account"
+	"github.com/NethermindEth/starknet.go/rpc"
+	"github.com/NethermindEth/starknet.go/utils"
+
+	setup "github.com/NethermindEth/starknet.go/examples/internal"
+)
+
+// More info: https://docs.starknet.io/architecture-and-concepts/accounts/universal-deployer/
+// NOTE : Please add in your keys only for testing purposes, in case of a leak you would potentially lose your funds.
+var (
+	someContractHash string = "0x046ded64ae2dead6448e247234bab192a9c483644395b66f2155f2614e5804b0" // The contract hash to be deployed (in this example, it's an ERC20 contract)
+	UDCAddress       string = "0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02bf" // UDC contract address
+	contractMethod   string = "deployContract"                                                     // UDC method to deploy a contract (from pre-declared contracts)
+)
+
+// Example succesful transaction created from this example on Sepolia
+// https://sepolia.voyager.online/tx/0x9bc6f6352663aafd71a9ebe1bde9c042590d8f3c8c265e5826274708cf0133
+
+func main() {
+	fmt.Println("Starting deployContractUDC example")
+
+	// Load variables from '.env' file
+	rpcProviderUrl := setup.GetRpcProviderUrl()
+	accountAddress := setup.GetAccountAddress()
+	accountCairoVersion := setup.GetAccountCairoVersion()
+	privateKey := setup.GetPrivateKey()
+	publicKey := setup.GetPublicKey()
+
+	// Initialize connection to RPC provider
+	client, err := rpc.NewProvider(rpcProviderUrl)
+	if err != nil {
+		panic(fmt.Sprintf("Error dialing the RPC provider: %s", err))
+	}
+
+	// Here we are converting the account address to felt
+	accountAddressInFelt, err := utils.HexToFelt(accountAddress)
+	if err != nil {
+		panic(err)
+	}
+
+	// Initialize the account memkeyStore (set public and private keys)
+	ks := account.NewMemKeystore()
+	privKeyBI, ok := new(big.Int).SetString(privateKey, 0)
+	if !ok {
+		panic("Fail to convert privKey to bitInt")
+	}
+	ks.Put(publicKey, privKeyBI)
+
+	fmt.Println("Established connection with the client")
+
+	// Set maxFee
+	maxfee, err := utils.HexToFelt("0x9184e72a000")
```

Comment:
I also had to raise the max fee here

Reply:
This was due to high gas price volatility. I fixed this in the new commit by updating the transaction's max fee field with the value returned by EstimateFee + 20%